src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -465,7 +465,7 @@ class InstanceKlass: public Klass {
   jushort nest_host_index() const { return _nest_host_index; }
   void set_nest_host_index(u2 i)  { _nest_host_index = i; }
   // dynamic nest member support
-  void set_nest_host(InstanceKlass* host, TRAPS);
+  void set_nest_host(InstanceKlass* host, JavaThread* current);
```

Comment:
I think we should drop the thread parameter for this method and nest_host_error and avoid the tramp data. We only need the current thread for a ResourceMark in a logging clause in set_nest_host, and for a ConstantPoolHandle in nest_host_error. Neither of these codepaths are hot so we can just manifest Thread::current() when needed there.
Then you can also remove the thread parameter from print_nest_host_error_on.

Reply:
You're right. I can remove a lot of thread arguments this way.

Reply:
Done.  Reran runtime/Nestmates tests and tier1.